sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs

```diff
@@ -874,6 +935,245 @@ public async Task GetBatchSettingWithAllFields()
+                Assert.GreaterOrEqual(settings.Length, 1);
+
+                foreach (ConfigurationSetting setting in settings)
+                {
+                    StringAssert.StartsWith("abc", setting.Key);
+                }
+            }
+            finally
+            {
+                await service.DeleteAsync(testSetting.Key);
+            }
+        }
+
+        [Test]
+        public async Task GetBatchSettingEndsWith()
+        {
+            ConfigurationClient service = GetClient();
+            ConfigurationSetting testSetting = CreateSetting("yzabc", "Test of ends with", "yzabc");
+            string endsWith = testSetting.Key.Substring(5);
+
+            try
+            {
+                await service.SetAsync(testSetting);
+
+                var selector = new SettingSelector($"*{endsWith}");
+
+                ConfigurationSetting[] settings = (await service.GetSettingsAsync(selector, CancellationToken.None).ToEnumerableAsync()).ToArray();
+
+                // There should be at least one key available.
+                Assert.GreaterOrEqual(settings.Length, 1);
+
+                foreach (ConfigurationSetting setting in settings)
+                {
+                    StringAssert.EndsWith(endsWith, setting.Key);
+                }
+            }
+            finally
+            {
+                await service.DeleteAsync(testSetting.Key);
+            }
+        }
+
+        [Test]
+        public async Task GetBatchSettingContains()
+        {
+            ConfigurationClient service = GetClient();
+            ConfigurationSetting testSetting = CreateSetting("yzabcde", "Contains abc", "yzabcde");
+
+            try
+            {
+                await service.SetAsync(testSetting);
+
+                var selector = new SettingSelector("*abc*");
+
+                ConfigurationSetting[] settings = (await service.GetSettingsAsync(selector, CancellationToken.None).ToEnumerableAsync()).ToArray();
+
+                // There should be at least one key available.
+                Assert.GreaterOrEqual(settings.Length, 1);
+
+                foreach (ConfigurationSetting setting in settings)
+                {
+                    StringAssert.Contains("abc", setting.Key);
+                }
+            }
+            finally
+            {
+                await service.DeleteAsync(testSetting.Key);
+            }
+        }
+
+        [Test]
+        public async Task GetBatchSettingsWithCommaInSelectorKey()
+        {
+            ConfigurationClient service = GetClient();
+
+            ConfigurationSetting abcSetting = new ConfigurationSetting("ab,cd", "comma in key");
+            ConfigurationSetting xyzSetting = new ConfigurationSetting("wx,yz", "comma in key");
+
+            try
+            {
+                await service.SetAsync(abcSetting);
+                await service.SetAsync(xyzSetting);
+
+                var selector = new SettingSelector(key:null);
```

Comment:
`new SettingSelector(key:null)` is equal to `new SettingSelector(SettingSelector.Any)`:
```cs
        public SettingSelector(string key, string label = default)
        {
            Keys = new List<string>
            {
                key ?? Any
            };

            Labels = new List<string>();
            if (label != null)
                Labels.Add(label);
        }
```
You probably want to have this:
```cs
var selector = new SettingSelector("ab,cd");
selector.Keys.Add("wz,yz");
```